### ISSUE TITLE
Failure [INSTALL_FAILED_NO_MATCHING_ABIS: Failed to extract native libraries, res=-113]

### ISSUE THREAD

**Issue by johnnyhonda** [bitcoinj/bitcoinj#1544]
Trying to install a basic Android app that uses bitcoinj.

I add bitcoinj to my `build.grade`

```
compile 'org.bitcoinj:bitcoinj-core:0.14.4'
```
I can build the project and it will run on the android emulator, but I cannot get the app to install on a physical device. I see the following error in android studio:

Failure [INSTALL_FAILED_NO_MATCHING_ABIS: Failed to extract native libraries, res=-113]

Any suggestions getting bitcoinj to install on a a physical android device?

**Comment by johnnyhonda** [comment]
Going to post this here in case anyone else is curious. You need to exclude the following native libs in order to run it on an arm device.

Add the following to your `build.gradle`

```
    packagingOptions {
        exclude 'lib/x86_64/darwin/libscrypt.dylib'
        exclude 'lib/x86_64/freebsd/libscrypt.so'
        exclude 'lib/x86_64/linux/libscrypt.so'
    }
```

**Comment by schildbach** [comment]
I think we can close this now.

**Comment by cyon1c** [comment]
Hi all! I'd like to re-open this issue as this does not actually fix the problem, but rather allows it to fail silently, and leaves several components of the library broken. I've detailed my issues [here](https://stackoverflow.com/questions/51368632/apk-fails-to-install-when-library-with-native-componentsbitcoinj-is-included) on stack overflow. This is an unusable solution if `bitcoinj` is included in a library project.

What is incredibly odd is that the library builds and deploys fine on my older projects, as well as new projects that include C++ support. 

I've also tried including as a jar, but that has it's own issues.
